`request failed: transport is closing` error on Delete operation for complex object (70Mb)

NeoFS version 0.19.0-40-g89b147e (latest master)
```
$ neofs-cli --rpc-endpoint s01.neofs.devenv:8080 --key L3vZAyGTNQkv18wrtvF6NdXCNa3KSRJs6QcbkXxotWKjEnyt2WdN object head --cid 9Qajp3VyoUsUkfGM5SpG6BuntNRrdkz5XyEDqmCV7U8y --oid HaQnS2NmWZePZv5veGoiemiHm5Dj7hgygzXEAHCxSPAD  
ID: HaQnS2NmWZePZv5veGoiemiHm5Dj7hgygzXEAHCxSPAD
CID: 9Qajp3VyoUsUkfGM5SpG6BuntNRrdkz5XyEDqmCV7U8y
Owner: NhdiKZ2GNJ2z4aDELkbpiTgzTRMTPMz9G6
CreatedAt: 1
Size: 70000000
HomoHash: 3366d512910d8d2247f6651ff34e28ca545efbcc8d07198fb0a41d62e8c641141ec87abd6516ed1d4c9306ac0ffe60f42e8dcb9bf562ae4d786bf465770e26d1
Checksum: 6a5be14f69ba0dd3d8f29626d0bc2d3c9950bcbbc67ee8b4ed32ba7c524258fb
Type: regular
Attributes:
  FileName=3a110fd5-51ea-43d5-962e-9493d688c081
  Timestamp=1621458514 (2021-05-20 00:08:34 +0300 MSK)
Split Header Signature:
  public key: 033634fb3f840c459647641fb79cbe2e5e574830a10716a5b993dbcbe6f237465b
  signature: 04b71e0c95eebbb700416257a8d36c6f2ee19d4e08d6a3fe970fff8bb01e7bc5ad06aebbd8f5aca2f4fc64b496e132bbdd2d3c4560669c1ffc37ede83bd463ec8f
 ```
  
 ```
$ neofs-cli --rpc-endpoint s01.neofs.devenv:8080 --key L3vZAyGTNQkv18wrtvF6NdXCNa3KSRJs6QcbkXxotWKjEnyt2WdN object delete --cid 9Qajp3VyoUsUkfGM5SpG6BuntNRrdkz5XyEDqmCV7U8y --oid HaQnS2NmWZePZv5veGoiemiHm5Dj7hgygzXEAHCxSPAD  
Error: sending the request failed: rpc error: code = Unavailable desc = transport is closing
Usage:
  neofs-cli object delete [flags]

Aliases:
  delete, del

Flags:
      --cid string   Container ID
  -h, --help         help for delete
      --oid string   Object ID

Global Flags:
      --address string        address of wallet account
      --bearer string         File with signed JSON or binary encoded bearer token
  -c, --config string         config file (default is $HOME/.config/neofs-cli/config.yaml)
  -k, --key --key new         private key in hex, WIF, NEP-2 or filepath (use --key new to generate key for request)
  -r, --rpc-endpoint string   remote node address (as 'multiaddr' or '<host>:<port>')
      --ttl uint32            TTL value in request meta header (default 2)
  -v, --verbose               verbose output
  -x, --xhdr strings          Request X-Headers in form of Key=Value
```

Seems like issue with search request forwarding.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xcd9fab]

goroutine 160 [running]:
github.com/nspcc-dev/neofs-node/pkg/services/object/search.(*clientWrapper).searchObjects(0xc00021bbb0, 0xc00030cf80, 0xc000543f80, 0x10f71e0, 0xc0005a1340, 0xc00020f310, 0xc0005a0540)
        github.com/nspcc-dev/neofs-node/pkg/services/object/search/util.go:79 +0x3b
github.com/nspcc-dev/neofs-node/pkg/services/object/search.(*execCtx).processNode(0xc00030cf80, 0x1113f40, 0xc0002cc380, 0xc00021bb10)
        github.com/nspcc-dev/neofs-node/pkg/services/object/search/remote.go:20 +0x233
github.com/nspcc-dev/neofs-node/pkg/services/object/search.(*execCtx).processCurrentEpoch(0xc00030cf80, 0xf99500)
        github.com/nspcc-dev/neofs-node/pkg/services/object/search/container.go:79 +0x22e
github.com/nspcc-dev/neofs-node/pkg/services/object/search.(*execCtx).executeOnContainer(0xc00030cf80)
        github.com/nspcc-dev/neofs-node/pkg/services/object/search/container.go:28 +0x1a4
github.com/nspcc-dev/neofs-node/pkg/services/object/search.(*execCtx).analyzeStatus(0xc00030cf80, 0xf88401)
        github.com/nspcc-dev/neofs-node/pkg/services/object/search/search.go:47 +0x9e
github.com/nspcc-dev/neofs-node/pkg/services/object/search.(*execCtx).execute(0xc00030cf80)
        github.com/nspcc-dev/neofs-node/pkg/services/object/search/search.go:32 +0x76
github.com/nspcc-dev/neofs-node/pkg/services/object/search.(*Service).Search(0xc0003122a0, 0x1114000, 0xc000543f80, 0x10f7000, 0xc0005a1320, 0xc00020f310, 0xc0005a0540, 0xc0003a2480, 0x1, 0x1, ...)
        github.com/nspcc-dev/neofs-node/pkg/services/object/search/search.go:21 +0xd2
github.com/nspcc-dev/neofs-node/pkg/services/object/delete.(*searchSvcWrapper).splitMembers(0xc0003122a0, 0xc0002feb60, 0x16, 0x0, 0x0, 0x0, 0xc0005a0da0)
        github.com/nspcc-dev/neofs-node/pkg/services/object/delete/util.go:90 +0x277
github.com/nspcc-dev/neofs-node/pkg/services/object/delete.(*execCtx).supplementBySplitID(0xc0002feb60, 0x40ce01)
        github.com/nspcc-dev/neofs-node/pkg/services/object/delete/exec.go:185 +0x91
github.com/nspcc-dev/neofs-node/pkg/services/object/delete.(*execCtx).collectMembers(0xc0002feb60, 0xc0007d8ff0)
        github.com/nspcc-dev/neofs-node/pkg/services/object/delete/exec.go:120 +0x46
github.com/nspcc-dev/neofs-node/pkg/services/object/delete.(*execCtx).formTombstone(0xc0002feb60, 0xf94c2a)
        github.com/nspcc-dev/neofs-node/pkg/services/object/delete/local.go:58 +0x3f1
github.com/nspcc-dev/neofs-node/pkg/services/object/delete.(*execCtx).executeLocal(0xc0002feb60)
        github.com/nspcc-dev/neofs-node/pkg/services/object/delete/local.go:11 +0x67
github.com/nspcc-dev/neofs-node/pkg/services/object/delete.(*execCtx).execute(0xc0002feb60)
        github.com/nspcc-dev/neofs-node/pkg/services/object/delete/delete.go:28 +0x63
github.com/nspcc-dev/neofs-node/pkg/services/object/delete.(*Service).Delete(0xc0003122d0, 0x1114000, 0xc000543f80, 0xc00020f310, 0xc0002858c0, 0x10f7020, 0xc0003124f0, 0x0, 0xc000100380)
        github.com/nspcc-dev/neofs-node/pkg/services/object/delete/delete.go:19 +0xc0
github.com/nspcc-dev/neofs-node/pkg/services/object/delete/v2.(*Service).Delete(0xc0003122d8, 0x1114000, 0xc000543f80, 0xc0005a0520, 0xc0002859d0, 0x1895dc0, 0xfbfd60)
        github.com/nspcc-dev/neofs-node/pkg/services/object/delete/v2/service.go:50 +0xed
main.(*objectSvc).Delete(0xc000301520, 0x1114000, 0xc000543f80, 0xc0005a0520, 0xc0002859c0, 0x0, 0x0)
        github.com/nspcc-dev/neofs-node/cmd/neofs-node/object.go:85 +0x4d
github.com/nspcc-dev/neofs-node/pkg/services/object.TransportSplitter.Delete(...)
        github.com/nspcc-dev/neofs-node/pkg/services/object/transport_splitter.go:107
github.com/nspcc-dev/neofs-node/pkg/services/object.(*ResponseService).Delete.func1(0x1114000, 0xc000543f80, 0xef92e0, 0xc0005a0520, 0x0, 0x8, 0xaf18d0, 0x1120340)
        github.com/nspcc-dev/neofs-node/pkg/services/object/response.go:122 +0x66
github.com/nspcc-dev/neofs-node/pkg/services/util/response.(*Service).HandleUnaryRequest(0xc00028a288, 0x1114000, 0xc000543f80, 0xef92e0, 0xc0005a0520, 0xc00061b6c0, 0xae28f0, 0x7f4eaf4beb78, 0x110fa00, 0xc0003124d0)
        github.com/nspcc-dev/neofs-node/pkg/services/util/response/unary.go:13 +0x5e
github.com/nspcc-dev/neofs-node/pkg/services/object.(*ResponseService).Delete(0xc000301560, 0x1114000, 0xc000543f80, 0xc0005a0520, 0xc00020f270, 0xd28e24, 0x0)
        github.com/nspcc-dev/neofs-node/pkg/services/object/response.go:120 +0x85
github.com/nspcc-dev/neofs-node/pkg/services/object.(*SignService).Delete.func1(0x1114000, 0xc000543f80, 0xef92e0, 0xc0005a0520, 0x18da9d0, 0x0, 0xc00061b798, 0xcc004c)
        github.com/nspcc-dev/neofs-node/pkg/services/object/sign.go:137 +0x66
github.com/nspcc-dev/neofs-node/pkg/services/util.(*SignService).HandleUnaryRequest(0xc0003122e0, 0x1114000, 0xc000543f80, 0xef92e0, 0xc0005a0520, 0xc00061b7e8, 0xcc0000, 0xc018888888, 0xc0003124b0, 0x7f4eaf4beb58)
        github.com/nspcc-dev/neofs-node/pkg/services/util/sign.go:119 +0x107
github.com/nspcc-dev/neofs-node/pkg/services/object.(*SignService).Delete(0xc000301580, 0x1114000, 0xc000543f80, 0xc0005a0520, 0xc00051b6a0, 0xc0005a0540, 0xc0005a05a0)
        github.com/nspcc-dev/neofs-node/pkg/services/object/sign.go:135 +0x86
github.com/nspcc-dev/neofs-node/pkg/services/object/acl.Service.Delete(0xc0003207d0, 0x1114000, 0xc000543f80, 0xc0005a0520, 0xad2a01, 0xc0005a05e0, 0xc00061ba78)
        github.com/nspcc-dev/neofs-node/pkg/services/object/acl/acl.go:294 +0x3f2
github.com/nspcc-dev/neofs-node/pkg/services/object.MetricCollector.Delete(0x111d920, 0xc0003207d0, 0x11237e0, 0xc0002b1c00, 0x1114000, 0xc000543f80, 0xc0005a0520, 0x0, 0x0, 0x0)
        github.com/nspcc-dev/neofs-node/pkg/services/object/metrics.go:117 +0xfa
github.com/nspcc-dev/neofs-node/pkg/network/transport/object/grpc.(*Server).Delete(0xc0004724b0, 0x1114000, 0xc000543f80, 0xc000303180, 0xc0004724b0, 0xc000543f80, 0xc000710ba0)
        github.com/nspcc-dev/neofs-node/pkg/network/transport/object/grpc/service.go:67 +0xa1
github.com/nspcc-dev/neofs-api-go/v2/object/grpc._ObjectService_Delete_Handler(0xec7400, 0xc0004724b0, 0x1114000, 0xc000543f80, 0xc0006ead80, 0x0, 0x1114000, 0xc000543f80, 0xc000624b00, 0x2a2)
        github.com/nspcc-dev/neofs-api-go@v1.26.0/v2/object/grpc/service.pb.go:3702 +0x217
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002fa820, 0x111f5a0, 0xc0007daa80, 0xc000380700, 0xc000306d80, 0x189d360, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.29.1/server.go:1082 +0x50a
google.golang.org/grpc.(*Server).handleStream(0xc0002fa820, 0x111f5a0, 0xc0007daa80, 0xc000380700, 0x0)
        google.golang.org/grpc@v1.29.1/server.go:1405 +0xccd
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000583e00, 0xc0002fa820, 0x111f5a0, 0xc0007daa80, 0xc000380700)
        google.golang.org/grpc@v1.29.1/server.go:746 +0xa1
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.29.1/server.go:744 +0xa1
```

--- a/pkg/services/object/get/exec.go
+++ b/pkg/services/object/get/exec.go
@@ -38,7 +38,7 @@ type execCtx struct {
 
 	curOff uint64
 
-	head, hash bool
+	head bool
 
 	curProcEpoch uint64
 
@@ -68,12 +68,6 @@ func headOnly() execOption {
 func withPayloadRange(r *objectSDK.Range) execOption {
 	return func(c *execCtx) {
 		c.prm.rng = r
-	}
-}
-
-func hashOnly() execOption {
-	return func(c *execCtx) {
-		c.hash = true
 	}
 }
 
@@ -150,10 +144,6 @@ func (exec *execCtx) ctxRange() *objectSDK.Range {
 
 func (exec *execCtx) headOnly() bool {
 	return exec.head
-}
-
-func (exec *execCtx) hashOnly() bool {
-	return exec.hash
 }
 
 func (exec *execCtx) netmapEpoch() uint64 {

--- a/pkg/services/object/get/get.go
+++ b/pkg/services/object/get/get.go
@@ -41,7 +41,7 @@ func (s *Service) GetRangeHash(ctx context.Context, prm RangeHashPrm) (*RangeHas
 			hash: util.NewSaltingWriter(h, prm.salt),
 		})
 
-		if err := s.getRange(ctx, rngPrm, hashOnly()); err != nil {
+		if err := s.getRange(ctx, rngPrm); err != nil {
 			return nil, err
 		}
 

--- a/pkg/services/object/get/util.go
+++ b/pkg/services/object/get/util.go
@@ -80,7 +80,7 @@ func (c *clientCacheWrapper) get(addr string) (getClient, error) {
 }
 
 func (c *clientWrapper) getObject(exec *execCtx) (*objectSDK.Object, error) {
-	if !exec.assembling && !exec.hashOnly() {
+	if !exec.assembling && exec.prm.forwarder != nil {
 		return exec.prm.forwarder(c.client)
 	}
 

--- a/pkg/services/object/search/util.go
+++ b/pkg/services/object/search/util.go
@@ -76,7 +76,16 @@ func (c *clientConstructorWrapper) get(addr string) (searchClient, error) {
 }
 
 func (c *clientWrapper) searchObjects(exec *execCtx) ([]*objectSDK.ID, error) {
-	return exec.prm.forwarder(c.client)
+	if exec.prm.forwarder != nil {
+		return exec.prm.forwarder(c.client)
+	}
+
+	return c.client.SearchObject(exec.context(),
+		&exec.prm.SearchObjectParams,
+		exec.prm.common.RemoteCallOptions(
+			util.WithNetmapEpoch(exec.curProcEpoch),
+			util.WithKey(exec.prm.common.PrivateKey()),
+		)...)
 }
 
 func (e *storageEngineWrapper) search(exec *execCtx) ([]*objectSDK.ID, error) {

